pkg/gofr/http/responder.go

```diff
@@ -35,9 +36,11 @@ func (r Responder) Respond(data interface{}, err error) {
 
 		return
 	default:
-		resp = response{
-			Data:  v,
-			Error: errorObj,
+		resp = response{Data: data, Error: errorObj}
+
+		// handling where an interface contains a nullable type with a nil value.
+		if isNil(data) {
+			resp = response{Error: errorObj}
 		}
 	}
```

Comment:
I'm unsure to understand why the omitempty is not enough?

Reply:
<img width="1127" alt="image" src="https://github.com/user-attachments/assets/51f301c5-acb9-4f76-ad44-8aaf3ca27f72">

In this case, it is not working. 

I'm not entirely sure why this is happening and might need to read more about it.